### PR TITLE
Aggregate functions: only create list if args exist

### PR DESCRIPTION
**Pull request by grooverdan**
Also always empty list rather than only if there is no error.

I submit this under the MCA.

### CODE REVIEW

File: sql/item_func.cc [hunk 1]
```diff
@@ -6330,17 +6330,20 @@ Item_func_sp::fix_fields(THD *thd, Item **ref)
...
     Item_sum_sp *item_sp;
     Query_arena *arena, backup;
     arena= thd->activate_stmt_arena_if_needed(&backup);
 
     if (arg_count)
+    {
+      List<Item> list;
+      list.empty();
+      for (uint i=0; i < arg_count; i++)
+        list.push_back(*(args+i));
```
Comment (svoj) [inline sql/item_func.cc]: Is it sophisticated way of writing `args[i]`?
Also it'd be nice to add space here: `i= 0`

**Comment by grooverdan** [comment]
Was just moving code that I saw was really only used for a single branch. Belatedly realised that an `arg_count` of 0 was an empty loop but there was still overhead in the loop construction. So I adjusted indenting (but not looking at other formatting), and it still passed the test case for it.

**Comment by svoj** [comment]
Good. Will you want to improve it further as I suggested or should I take care of it?

**Comment by grooverdan** [comment]
if you can take care of it that would be appreciated. Or pass it to @cvicentiu who signed off the original commit :-)

**Comment by svoj** [comment]
Pushed a3ccad0f21e48e30c98f91528e3fcbbf2c1d509a and 8a3a332bc083d4900431da65e2375a1f89ad02f6.

**Comment by grooverdan** [comment]
thanks for the merge and cleanup.

**Pull request closed without merging**